### ISSUE TITLE
!mock breaks YouTube links

### ISSUE THREAD

**Issue by treyzania** [edward-shen/02#5]
This applies to `youtu.be` links, but also may include the query part of full `youtube.com` links (the part after the `?`).  It might be worthwhile to just disable the mocking for all links.

**Comment by edward-shen** [comment]
Is this really a problem though? If you're mocking someone, you don't really care about what they're linking to, just that they said something stupid.

**Comment by treyzania** [comment]
Yes you do.  The function should be closed with respect to its input, so if links change you're breaking out of the set.

**Comment by edward-shen** [comment]
The function *is* currently closed with respect to its input. It simply doesn't recognise higher order structures (e.g. links, grammar, proper english capitalization, etc.) as valid input. It automatically maps higher order functions to input it accepts, but it's a lossy mapping.

Regardless, it still achieves its purpose. If you *really* have an issue with this, feel free to PR.

**Comment by treyzania** [comment]
(embed strings with URLs it into the type system)

**Comment by edward-shen** [comment]
(create a parser to do that for me)

**Comment by treyzania** [comment]
(in racket this is just
